### PR TITLE
Final steps before the release of @excalidraw/excalidraw

### PR DESCRIPTION
**Pull request by ad1992**
- [x] Added Readme
- [x] Added changelog
- [x] Updated version

### CODE REVIEW

File: src/packages/excalidraw/README.md [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+| [onPointerUpdate](#onPointerUpdate)         | Function                                                                                                                                                                                                                                                                                                                 |                                | Callback triggered when mouse pointer is updated.                                                                                                          |
+
+<a name="width"></a>
+
+#### width
+
+This props defines the width of the Excalidraw component. Defaults to `window.innerWidth` if not passed.
+
+<a name="height"></a>
```
Comment (lipis) [inline src/packages/excalidraw/README.md:111]:
```suggestion
```html
<a name="height"></a>
```
```

File: src/packages/excalidraw/README.md [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+
+#### height
+
+This props defines the height of the Excalidraw component. Defaults to `window.innerHeight` if not passed.
+
+<a name="offsetLeft"></a>
```
Comment (lipis) [inline src/packages/excalidraw/README.md:117]:
```html

File: src/packages/excalidraw/README.md [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+
+This helps to load Excalidraw with `initialData`.
+It must be an object or a [promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/Promise) which resolves to an object containing the below optional fields.
+
+| name     | type                                                                                                                                    |
+| -------- | --------------------------------------------------------------------------------------------------------------------------------------- |
+| elements | [ExcalidrawElement []](https://github.com/excalidraw/excalidraw/blob/4c90ea5667d29effe8ec4a115e49efc7c340cdb3/src/element/types.ts#L59) |
+| appState | [AppState](https://github.com/excalidraw/excalidraw/blob/4c90ea5667d29effe8ec4a115e49efc7c340cdb3/src/types.ts#L33)                     |
+
+```javascript
```
Comment (lipis) [inline src/packages/excalidraw/README.md:157]:
```suggestion
```json
```

File: src/packages/excalidraw/README.md [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+| history                          | { clear: () => void }                                                                                                                             | This is the history API. `history.clear()` will clear the history                                                                                                                                                                                |
+| setScrollToCenter                | ([ExcalidrawElement []](https://github.com/excalidraw/excalidraw/blob/4c90ea5667d29effe8ec4a115e49efc7c340cdb3/src/element/types.ts#L59)) => void | sets the elements to center                                                                                                                                                                                                                      |
+
+<a name="readyPromise"></a>
+
+#### readyPromise
```
Comment (lipis) [inline src/packages/excalidraw/README.md:214]:
```suggestion
#### `readyPromise`
```

File: src/packages/excalidraw/README.md [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+
+<a name="user"><a/>
+
+#### user
+
+This is the user name which shows during collaboration. Defaults to `{name: ''}`.
+
+<a name="excalidrawRef"></a>
+
+#### excalidrawRef
```
Comment (lipis) [inline src/packages/excalidraw/README.md:196]:
```suggestion
#### `excalidrawRef`
```

File: src/packages/excalidraw/README.md [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+
+<a name="offsetTop"></a>
+
+#### offsetTop
+
+This prop defines top position relative to which Excalidraw should be rendered
+
+<a name="onChange"></a>
+
+#### onChange
```
Comment (lipis) [inline src/packages/excalidraw/README.md:131]:
```suggestion
#### `onChange`
```

File: src/packages/excalidraw/README.md [hunk 7]
```diff
@@ -0,0 +1,216 @@
...
+| history                          | { clear: () => void }                                                                                                                             | This is the history API. `history.clear()` will clear the history                                                                                                                                                                                |
+| setScrollToCenter                | ([ExcalidrawElement []](https://github.com/excalidraw/excalidraw/blob/4c90ea5667d29effe8ec4a115e49efc7c340cdb3/src/element/types.ts#L59)) => void | sets the elements to center                                                                                                                                                                                                                      |
+
+<a name="readyPromise"></a>
+
+#### readyPromise
+
+const excalidrawRef = { current: { readyPromise: [resolvablePromise](https://github.com/excalidraw/excalidraw/blob/4c90ea5667d29effe8ec4a115e49efc7c340cdb3/src/utils.ts#L312) }
```
Comment (lipis) [inline src/packages/excalidraw/README.md:216]:
```
```js
```

File: src/packages/excalidraw/README.md [hunk 8]
```diff
@@ -0,0 +1,216 @@
...
+
+You can use npm
+
+```
+npm install react react-dom @excalidraw/excalidraw
+```
+
+or via yarn
+
+```cassandraql
```
Comment (lipis) [inline src/packages/excalidraw/README.md:15]:
```suggestion
```
```

Comment (lipis) [inline src/packages/excalidraw/README.md:15]: I think it's either `bash` or nothing in this case.. what's casandraql?

File: src/packages/excalidraw/README.md [hunk 9]
```diff
@@ -0,0 +1,216 @@
...
+| ------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ | ------------------------------ | ---------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| [width](#width)                             | Number                                                                                                                                                                                                                                                                                                                   | window.innerWidth              | The width of Excalidraw component                                                                                                                          |
```
Comment (lipis) [inline src/packages/excalidraw/README.md:93]:
```suggestion
| [`width`](#width)                           | Number                                                                                                                                                                                                                                                                                                                   | `window.innerWidth`            | The width of Excalidraw component                                                                                                                          |
```

File: src/packages/excalidraw/README.md (hunk 8 above)
Comment (ad1992) [reply src/packages/excalidraw/README.md]: This got introduced by autocomplete in my editor  :D (most probably it's for Apache Cassandra ), I have removed it.

File: src/packages/excalidraw/README.md [hunk 10]
```diff
@@ -0,0 +1,286 @@
...
+yarn add react react-dom @excalidraw/excalidraw
+```
+
+After installation you will see a folder `excalidraw-assets` in `dist` directory which contains the assets needed for this app.
+
+Move the folder `excalidraw-assets` to the path where your assets are served. In the example its served from `public/excalidraw-assets`
+
+### Demo
+
+[Try here](https://codesandbox.io/s/excalidraw-ehlz3)
```
Comment (lipis) [inline src/packages/excalidraw/README.md:25]:
```suggestion
[Try here](https://codesandbox.io/s/excalidraw-ehlz3).
```

File: src/packages/excalidraw/README.md [hunk 11]
```diff
@@ -0,0 +1,286 @@
...
+| [`width`](#width)                             | Number                                                                                                                                                                                                                                                                                                                         | window.innerWidth  | The width of Excalidraw component                                                                                                                          |
+| [`height`](#height)                           | Number                                                                                                                                                                                                                                                                                                                         | window.innerHeight | The height of Excalidraw component                                                                                                                         |
```
Comment (lipis) [inline src/packages/excalidraw/README.md:134]: `window.innerHeight`, `window.innerWidth`, in general anything that is code should be code

File: src/packages/excalidraw/README.md [hunk 12]
```diff
@@ -0,0 +1,286 @@
...
+
+This prop defines left position relative to which Excalidraw should be rendered. Defaults to 0 if not passed.
+
+<a name="offsetTop"></a>
+
+#### `offsetTop`
+
+This prop defines top position relative to which Excalidraw should be rendered. Defaults to 0 if not passed.
+
+<a name="onChange"></a>
```
Comment (lipis) [inline src/packages/excalidraw/README.md:169]: what are these? for the links? GitHub supports links for the headers by default

File: src/packages/excalidraw/README.md [hunk 13]
```diff
@@ -0,0 +1,286 @@
...
+| -------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| ready                            | boolean                                                                                                                                           | This is set to true once Excalidraw is rendered                                                                                                                                                                                                  |
```
Comment (lipis) [inline src/packages/excalidraw/README.md:243]:
```suggestion
| `ready`                          | `boolean`                                                                                                                                         | This is set to true once Excalidraw is rendered                                                                                                                                                                                                  |
```

File: src/packages/excalidraw/CHANGELOG.MD [hunk 14]
```diff
@@ -0,0 +1,5 @@
+# Changelog
+
+## 0.1.0
+
+First release of `@excalidraw/excalidraw`
```
Comment (lipis) [inline src/packages/excalidraw/CHANGELOG.MD:5]: Is that going to be manually updated?

**Comment by lipis** [comment]
I haven't followed this package in general, but is there a plan to add a GitHub action to automatically publish on npm.. 

How the releases are going to work? Would be nice to automate as much as possible.

File: src/packages/excalidraw/CHANGELOG.MD (hunk 14 above)
Comment (ad1992) [reply src/packages/excalidraw/CHANGELOG.MD:5]: Not really we can automate this with a script that just fetches the commits after the last release and adds it here.

**Comment by ad1992** [comment]
> I haven't followed this package in general, but is there a plan to add a GitHub action to automatically publish on npm..
> 
> How the releases are going to work? Would be nice to automate as much as possible.

Yes, we should automate releases, I was thinking of using something like [semantic-release](https://github.com/semantic-release/semantic-release) which helps in creating release notes as well so we need not worry about changelog or deciding what should be the next version, etc.  We can discuss this with the team after the initial release.

**Comment by dwelle** [comment]
> Yes, we should automate releases, I was thinking of using something like [semantic-release](https://github.com/semantic-release/semantic-release) which helps in creating release notes as well so we need not worry about changelog or deciding what should be the next version, etc. We can discuss this with the team after the initial release.

That workflow still relies on people naming their commits correctly.

I haven't filly reviewed the readme, but I trust @ad1992 did a good job, and we can always make tweaks in the future. Feel free to merge.

**Pull request merged**